values.yaml

```diff
@@ -25,6 +25,26 @@ ingress:
   # Name of tls secret to use on ingress
   tlsSecretName: ~
 
+  # Annotations always injected when configuring Ingress ojbect
+  annotations: |-
```

Comment:
What happens if this is not intended correctly (4 spaces, but instead say 2) - does that break the yaml?

If so making these lists, and using `range` in the template is probably safer.

Reply:
`Using |-` will strip leading indents anyway, so what happens if this is just set to:

```
    annotations: "kubernetes.io/ingress.class: x"
```

say?

Reply:
@ashb I'm still not quite certain what the issue is here?

Reply:
By making this a string it's harder to get right -- the indentation matters here more than just for YAML here -- it has to be correct in the end result, once the template is rendered, doesn't it?)

If instead the structure was:

```yaml
  annotations:
       kubernetes.io/ingress.class: "{{ .Values.ingress.class }}"
```

i.e. a dict,  then it's "harder" to get wrong -- if this file is valid YAML the end result when the annotation template is rendered is much more likely to be valid YAML.

Reply:
I just don't think that's flexible enough... we have cases of conditional values, sub-values etc. and I'm not sure we could handle all of those in a dict. Though I guess we could offer both string AND dict? @schnie what do you think?